contracts/contracts/rem.swap/include/rem.swap/rem.swap.hpp

```diff
@@ -0,0 +1,210 @@
+         *
+         * @param user - the owner account to execute the finish action for,
+         * @param receiver - the account to be swap finished to,
+         * @param trx_id - the transaction transfer id in Ethereum blockchain,
+         * @param quantity - the quantity of tokens to be swaped,
+         * @param sign - the signature that sign by swap-key with data : name receiver account,
+         * @param swap_init_time - the timestamp transfer transaction in Ethereum blockchain.
+         */
+        [[eosio::action]]
+        void finish( const name& user, const name& receiver, const string& trx_id,
+                     asset& quantity, const signature& sign, const time_point& swap_init_time );
+
+
+        /**
+         * Finish token swap with ERC20 REM.
+         *
+         * @details Finish already approved token swap ( from ERC20 - > Remchain ).
+         *
+         * @param user - the owner account to execute the finish action for,
+         * @param receiver - the account to be swap finished to,
+         * @param trx_id - the transaction transfer id in Ethereum blockchain,
+         * @param quantity - the quantity of tokens to be swaped,
+         * @param sign - the signature that sign by swap-key with data : name receiver account,
+         * owner_key, public_key,
+         * @param swap_init_time - the timestamp transfer transaction in Ethereum blockchain,
+         * @param owner_key - owner key account to create,
+         * @param active_key - active key account to create.
+         */
+        [[eosio::action]]
+        void finishnewacc( const name& user, const name& receiver, const string& trx_id,
+                            asset& quantity, const signature& sign, const time_point& swap_init_time,
+                            const public_key owner_key, const public_key active_key                   );
+
+
+        [[eosio::on_notify("eosio.token::transfer")]]
+        void initerc( name from, name to, asset quantity, string memo );
+
+        using init_swap_action = eosio::action_wrapper<"init"_n, &swap::init>;
+        using finish_swap_action = eosio::action_wrapper<"finish"_n, &swap::finish>;
+        using finish_swap_and_create_acc_action = eosio::action_wrapper<"finishwithacc"_n, &swap::finish>;
+
+    private:
+        enum class swap_status: uint8_t {
+            INIT = 0,
+            FINISH = 1
+        };
+
+        const asset create_account_fee = {20, symbol("REM", 4)};
+        const string remchain_id = "cf057bbfb72640471fd910bcb67639c22df9f92470936cddc1ade0e2f2e7dc4f";
+        const time_point swap_lifetime = time_point(seconds(15552000)); // 180 days
+
+        struct approval {
+            permission_level level;
+            time_point       time;
+        };
+
+        struct [[eosio::table]] swap_data {
+            uint64_t                key;
+            string                  trx_id;
+            checksum256             swap_hash;
+            time_point              swap_init_time;
```

Comment:
```suggestion
            block_timestamp              swap_init_time;
```
sizeof(block_timestamp) == 4
sizeof(time_point) == 8